packages/web/src/components/social-proof/SocialProof.tsx

```diff
@@ -0,0 +1,227 @@
+            onClick={onInstagramClick}
+            onSuccess={onInstagramLogin}
+            onFailure={(error: Error) => onFailure('instagram', error)}
+          />
+        )}
+      </div>
+      {error && (
+        <div className={styles.error}>
+          <IconValidationX className={styles.validationIcon} />
+          {error}
+        </div>
+      )}
+    </div>
+  )
+}
+
+const LoadingBody = () => {
+  return (
+    <div className={styles.container}>
+      <LoadingSpinner className={styles.loadingContainer} />
+    </div>
+  )
+}
+
+type SocialProofProps = {
+  onSuccess: () => void
+}
+
+/**
+ * A modal that verifies social proof of a user.
+ * When proof is made, the modal closes. There are no
+ * callbacks so that a server may determine proof.
+ */
+const SocialProof = ({ onSuccess }: SocialProofProps) => {
+  const dispatch = useDispatch()
+  const onInstagramLogin = useCallback(
+    (uuid: string, profile: InstagramProfile) => {
+      dispatch(instagramLoginAction({ uuid, profile }))
+    },
+    [dispatch]
+  )
+  const onTwitterLogin = useCallback(
+    (uuid: string, profile: TwitterProfile) => {
+      dispatch(twitterLoginAction({ uuid, profile }))
+    },
+    [dispatch]
+  )
+
+  const [error, setError] = useState('')
+  const [status, setStatus] = useState('')
+  const [isOpen, setIsOpen] = useModalState('SocialProof')
+  const handle = useSelector(getUserHandle)
+
+  const onClick = useCallback(() => setStatus(Status.LOADING), [setStatus])
+  const record = useRecord()
+  const onFailure = useCallback(
+    (type: 'instagram' | 'twitter', error: Error) => {
+      if (!handle) return
+
+      setError(messages.failure)
+      setStatus(Status.ERROR)
+
+      const trackEvent: TrackEvent = make(Name.SOCIAL_PROOF_ERROR, {
+        handle,
+        type,
+        error: error.message
+      })
+      record(trackEvent)
+    },
+    [record, setError, setStatus, handle]
+  )
+
+  const instagramLogin = useCallback(
+    (uuid: string, profile: InstagramProfile) => {
+      if (!handle) return
+
+      setStatus(Status.SUCCESS)
+      onInstagramLogin(uuid, profile)
+
+      const trackEvent: TrackEvent = make(Name.SOCIAL_PROOF_SUCCESS, {
+        handle,
+        type: 'instagram',
+        screenName: profile.username
+      })
+      record(trackEvent)
+    },
+    [record, handle, onInstagramLogin]
+  )
+
+  const twitterLogin = useCallback(
+    (uuid: string, profile: TwitterProfile) => {
+      if (!handle) return
+
+      setStatus(Status.SUCCESS)
+      onTwitterLogin(uuid, profile)
+
+      const trackEvent: TrackEvent = make(Name.SOCIAL_PROOF_SUCCESS, {
+        handle,
+        type: 'twitter',
+        screenName: profile.screen_name
+      })
+      record(trackEvent)
+    },
+    [record, handle, onTwitterLogin]
+  )
+
+  useEffect(() => {
+    if (status === Status.SUCCESS) {
+      onSuccess()
+    }
+  }, [onSuccess, status])
+
+  let body = null
+  if (status === Status.LOADING || handle === null) {
```

Comment:
so if the handle is missing that means the selector has triggered i fetch for it? just checking that we dont get infinite loading or something